aws_crt_common_ws/aws-crt-common-sys/build.rs

```diff
@@ -0,0 +1,337 @@
+use aws_crt_c_flags::CRTModuleBuildInfo;
+use std::borrow::{Borrow, BorrowMut};
+use std::path::Path;
+
+fn load_compiler_configuration(build_config: &mut CRTModuleBuildInfo) {
+    let mut config_file_str: String = "#ifndef AWS_COMMON_CONFIG_H
+#define AWS_COMMON_CONFIG_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+/*
+ * This header exposes compiler feature test results determined during cmake
+ * configure time to inline function implementations. The macros defined here
+ * should be considered to be an implementation detail, and can change at any
+ * time.
+ */\n"
+        .to_string();
+
+    build_config.add_public_define("AWS_NO_STATIC_IMPL", "1");
+    config_file_str = format!("{}#define AWS_NO_STATIC_IMPL 1\n", config_file_str);
+
+    #[target_feature(enable = "avx2")]
+    {
+        build_config.add_private_define("HAVE_AVX2_INTRINSICS", "1");
+        build_config.add_private_define("HAVE_MM256_EXTRACT_EPI64", "1");
+
+        if build_config.get_toolchain().get_compiler().is_like_msvc() {
```

Comment:
This is something that should be encapsulated like the cflags stuff above, we're going to end up with nests of this all over otherwise.